src/server/frontend_xwayland/xwayland_server.cpp

```diff
@@ -107,62 +107,45 @@ void mf::XWaylandServer::spawn()
     enum { server, client, size };
     int wl_client_fd[size], wm_fd[size];
 
-    int xserver_spawn_tries = 0;
     std::unique_lock<decltype(spawn_thread_mutex)> lock{spawn_thread_mutex};
 
-    do
-    {
-        // Try 5 times
-        if (xserver_spawn_tries >= 5) {
-          mir::log_error("Xwayland failed to start 5 times in a row, disabling Xserver");
-          return;
-        }
-        spawn_thread_xserver_status = STARTING;
-        xserver_spawn_tries++;
-
-        if (socketpair(AF_UNIX, SOCK_STREAM | SOCK_CLOEXEC, 0, wl_client_fd) < 0)
-        {
-            // "Shouldn't happen" but continuing is weird.
-            mir::fatal_error("wl connection socketpair failed");
-            return; // doesn't reach here
-        }
+    spawn_thread_xserver_status = STARTING;
 
-        if (socketpair(AF_UNIX, SOCK_STREAM | SOCK_CLOEXEC, 0, wm_fd) < 0)
-        {
-            // "Shouldn't happen" but continuing is weird.
-            mir::fatal_error("wm fd socketpair failed");
-            return; // doesn't reach here
-        }
+    if (socketpair(AF_UNIX, SOCK_STREAM | SOCK_CLOEXEC, 0, wl_client_fd) < 0)
+    {
+        // "Shouldn't happen" but continuing is weird.
+        mir::fatal_error("wl connection socketpair failed");
```

Comment:
General question. Should this be a `mir::fatal_error`? When should something be a `mir::fatal_error`, and when should something just result in an exception being thrown?

*My* understanding is that the existence of `mir::fatal_error` is to improve debugging - we want to hit the `abort()` and get the backtrace without any other cleanup. I don't think that applies here; if `socketpair` fails then it's obvious from `errno` (which we're throwing away here :stuck_out_tongue_winking_eye: ).

So I have a weak preference for just throwing a `std::system_error` here.